src/mixins/embed.js

```diff
@@ -45,26 +45,28 @@ const debouncedOnClick = debounce((that) => {
   const count = that.likeCount - that.likeSent;
   that.likeSent += count;
   if (count > 0) {
-    apiPostLikeButton(
-      that.id,
-      count,
-      {
+    if (that.iscnId) {
+      apiPostLikeButtonByIscnId(that.iscnId, count, {
+        isCookieSupport: that.hasCookieSupport,
+        ...that.apiMetadata,
+      });
+    } else {
+      apiPostLikeButton(that.id, count, {
         referrer: that.referrer,
         isCookieSupport: that.hasCookieSupport,
         ...that.apiMetadata,
-      },
-    );
+      });
+    }
```

Comment:
Can we actually overload `apiPostLikeButton` that we use `iscn` as first param, then put in `iscnId` in the third object param?
it seems that the url structure of both are actually very similar, and we can skip a lot of `if else` in this PR if we use a common interface for both